src/vm/datatypes/strings.c

```diff
@@ -724,6 +724,87 @@ static Value wrapString(DictuVM *vm, int argCount, Value *args) {
     return OBJ_VAL(takeString(vm, temp, strlen(temp)));
 }
 
+static Value snakeToCamelCaseString(DictuVM *vm, int argCount, Value *args) {
+    if (argCount != 0) {
+        runtimeError(vm, "snakeToCamelCase() takes no arguments (%d given)", argCount);
+        return EMPTY_VAL;
+    }
+
+    ObjString *string = AS_STRING(args[0]);
+
+    if (string->length < 3) {
+        return OBJ_VAL(string);
+    }
+    if (strchr(string->chars, '_') == NULL) {
+        return OBJ_VAL(string);
+    }
+
+    char *temp = ALLOCATE(vm, char, string->length + 1);
+    
+    int i, j;
+    for (i = j = 0; i < string->length; i++, j++) {
+        if (string->chars[i] == '_') {
+            temp[j] = toupper(string->chars[i+1]);
```

Comment:
This is super dangerous, it does assumptions about the length of string, if a string ends with "_" this will append the null byte as content to temp, if toupper handles that

Reply:
Yeah valid point, probably need a length check here